tests/integration/requests_tests.py

```diff
@@ -269,12 +269,10 @@ def test_prometheus_http_client_metrics_inside_local_span(method, http_server):
     logger.info("Registered observer with baseplate.")
 
     # we need to clear metrics between test runs from the parametrize annotations
-    REGISTRY._names_to_collectors["http_client_requests_total"].clear()
-    REGISTRY._names_to_collectors["http_client_latency_seconds"].clear()
-    REGISTRY._names_to_collectors["http_client_active_requests"].clear()
-    REGISTRY._names_to_collectors["local_span_latency_seconds"].clear()
-    REGISTRY._names_to_collectors["local_span_requests_total"].clear()
-    REGISTRY._names_to_collectors["local_span_active_requests"].clear()
+    for collector_name in REGISTRY._names_to_collectors:
+        reg = REGISTRY._names_to_collectors[collector_name]
+        if hasattr(reg, "clear") and callable(getattr(reg, "clear")):
```

Comment:
Apparently there's a garbage collector in this list that doesn't have `clear()`

This belongs in a setup method but I didn't feel like refactoring this whole file for a test class that has a setup method. The reason this changed is that `local` spans no longer had a metric created for them at this point. This also helps future runs to ensure they're all cleared.